chore(roxctl): hide non-required global flags where applicable

dhaus67 opened the pull request:
## Description

This PR refactors the currently global flags on the `roxctl` command.

Instead of exposing flags related to central connectivity (e.g. endpoint, protocol choice etc.) to _all_ roxctl commands, instead only expose it to commands which use it (or respectively, their subcommands).

The reason for this is that the by default created persistent flags about central connectivity (endpoint, grpc type etc.), may become confusing for clients using subtree of commands where central connectivity is not required.

Additionally, it lead to some of the flags to be less discoverable that are defined as persistent flags (i.e. the config map flags, the shared flags for auth provider types for `roxctl declarative-config create`).

Additionally all scripts have been adjusted which specified `roxctl -e <>`, since the flag now must be passed _after_ the command which provides the global flag has been passed (e.g. `roxctl central -e <>`

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

- see CI

dhaus67 committed:
Hide connectivity flags for selected commands.

0x656b694d commented:
Wouldn't it be possible to change the flags hierarchy instead of hiding some flags? Like instead of
```
        Base set with central connectivity
        /      \
declarative   central
```
implement this:
```
        Base set without central connectivity
        /      \
declarative   base set with central connectivity
                  \
                  central
```

janisz reviewed (approved):
How about adding bats test to verify if flags are not displayed?

dhaus67 commented:
/retest

dhaus67 commented:
@janisz have a look again, I added:
- bats test which check whether only a single global flag is given (i.e. no-color) or multiple global flags with a partial match to the central flags (I didn't put _all_ global flags, since some command specify i.e. timeout as a global flag, and maintaining this seems also like a whole lot of pain).
- I changed the way the central connectivity flags are added, and only added it to the subcommands that require it (i.e. `central, cluster, collector, deployment, image, scanner, sensor`). Only these will have the connectivity flags and the rest won't get them to begin with, thus ensuring we don't need to hide them like @0x656b694d suggested. I quickly glanced over the commands, and we should be good to leave out the central connectivity flags on `completion, generate, helm, version, declarative config`.

0x656b694d commented on roxctl/declarativeconfig/create/auth_provider.go:
```diff
@@ -19,6 +19,21 @@ import (
 )
 
+var (
+	persistentFlagsToShow = []string{
```
Unused?

dhaus67 commented:
/test

dhaus67 commented:
/test all

parametalol reviewed (approved):
LGTM

dhaus67 commented:
/retest

dhaus67 commented:
Note for reviewers: the original approach of only adding the connectivity flags to subcommands which require them doesn't work, this would create a breaking change.
Things like `roxctl -e "<central-address>" central ...` wouldn't work anymore. Sadly, this is also the _exact_ example we use within docs. So, reverted the changes back to the original implementation to opt-out / hide the commands for specific subcommands.
